### ISSUE TITLE
release 2.0 milestones

### ISSUE THREAD

**Issue by clay-arras** [rushlite/rushlite#89]
milestones I want to hit for release 2.0:

**address friction log:** 
- .sum() doesn't work, need to specify axis
- need to cast w.grad to a variable first (actual: b.data -= 0.001 * rsl.Variable(b.grad), expected: b -= 0.001 * b.grad)
- torch.no_grad() equivalent
- ...

https://docs.google.com/document/d/1rPGKS5W1hrtrHEw9LcgGY8YAlTHBiR0x6HFx3WtwTh0/edit?usp=sharing

### get a working implementation of transformer architecture
**blockers (for now)**
- batched matmul
- slicing
- nn.Embedding equivalent
- torch.cat or vstack, some way to stack heads

https://github.com/clay-arras/llm-sons/blob/main/src/gpt.py

### miscellaneous cool features
add scheduler / orchestrator? https://horace.io/brrr_intro.html see OVERHEAD section
#80 optional
#74 for sure
